syncmaster/worker/settings/__init__.py

```diff
@@ -1,22 +1,70 @@
 # SPDX-FileCopyrightText: 2023-2024 MTS PJSC
 # SPDX-License-Identifier: Apache-2.0
+import importlib
+import os
+
+from pydantic import Field
 from pydantic.types import ImportString
+from pydantic_settings import BaseSettings
 
-from syncmaster.settings import SyncmasterSettings
+from syncmaster.settings import (
+    CredentialsEncryptionSettings,
+    DatabaseSettings,
+    LoggingSettings,
+    RabbitMQSettings,
+)
 
 
-class WorkerSettings(SyncmasterSettings):
+class WorkerSettings(BaseSettings):
     """Celery worker settings.
 
     Examples
     --------
 
     .. code-block:: bash
 
-        SYNCMASTER__WORKER__LOGGING__PRESET=colored
+        SYNCMASTER__WORKER__LOG_URL_TEMPLATE="https://grafana.example.com?correlation_id={{ correlation_id }}&run_id={{ run.id }}"
     """
 
-    CORRELATION_CELERY_HEADER_ID: str = "CORRELATION_CELERY_HEADER_ID"
-    LOG_URL_TEMPLATE: str = ""
+    LOG_URL_TEMPLATE: str = Field(
+        "",
+        description="URL template for logging",
+    )
+    CORRELATION_CELERY_HEADER_ID: str = Field(
+        "CORRELATION_CELERY_HEADER_ID",
+        description="Header ID for correlation in Celery",
+    )
+    CREATE_SPARK_SESSION_FUNCTION: ImportString = Field(
+        "syncmaster.worker.spark.get_worker_spark_session",
+        description="Function to create Spark session for worker",
+    )
+
+
+class WorkerAppSettings(BaseSettings):
+    TZ: str = "UTC"
+
+    database: DatabaseSettings = Field(description="Database settings")
+    broker: RabbitMQSettings = Field(description="Broker settings")
+    logging: LoggingSettings = Field(default_factory=LoggingSettings, description="Logging settings")
+    worker: WorkerSettings = Field(default_factory=WorkerSettings, description="Worker-specific settings")
+    encryption: CredentialsEncryptionSettings = Field(
+        default_factory=CredentialsEncryptionSettings,
+        description="Settings for encrypting credential data",
+    )
+
+    class Config:
+        env_prefix = "SYNCMASTER__"
+        env_nested_delimiter = "__"
+
+
+def get_worker_settings() -> WorkerAppSettings:
+    # TODO: add to worker documentation
+    worker_settings_path = os.environ.get("WORKER_SETTINGS", None)
 
-    CREATE_SPARK_SESSION_FUNCTION: ImportString = "syncmaster.worker.spark.get_worker_spark_session"
+    if worker_settings_path:
+        module_name, class_name = worker_settings_path.rsplit(".", 1)
+        module = importlib.import_module(module_name)
+        settings_class = getattr(module, class_name)
+    else:
+        settings_class = WorkerAppSettings
+    return settings_class()
```

Comment:
What's the point of this code?

Reply:
now we can provide worker settings via env
as we start worker via celery I cannot see any clear way to provide path to worker settings class like:
`python -m celery -A syncmaster.worker.config.celery worker --worker-settings  python -m celery syncmaster.worker.settings.WorkerAppSettings`

because celery will also try to parse this command and fail with:

```
Usage: python -m celery [OPTIONS] COMMAND [ARGS]...
Try 'python -m celery --help' for help.

Error: Invalid value for '-A' / '--app': 
```

what is best approach to provide custom worker settings class in worker application?

Reply:
Why to we need to `provide path to worker settings class`?

Reply:
in case we want to override default settings (for example user will have multiple workers and want different configurations for them), I thought that was the purpose of providing function with getting worker settings

Reply:
Different workers will start in different docker containers, each one could have its own .env file.

> I thought that was the purpose of providing function with getting worker settings

I haven't proposed to add this function in the first place.

What I mentioned in previous PRs is that creating worker settings during import of `syncmaster.worker.config.celery` module is not the best way to inject settings to the application. It prevents us from changing these settings in runtime (like in worker unit tests where we could require change settings between different tests).

It would be better, if we had worker factory function like this:
```python
def create_worker(settings: WorkerAppSettings):
  return Celery(...)
```
So we could call factory function and pass settings here to get a worker instance. But I'm not sure if Celery support this kind of usage.

Reply:
updated creating of celery app

Reply:
Integration tests are failing, is worker alive?